src/mutable-prim-array-types.hs

```diff
@@ -1,27 +1,31 @@
 -- MOVE THIS TO A DIFFERENT LOCATION
 
-import GHC.Exts
-
+import Control.Monad (when, void)
 import Control.Monad.Primitive
 import Data.Primitive.Types
-
-import Prelude hiding (length, unlines)
-
 import Streamly.Internal.Data.Fold.Types (Fold(..))
 import Streamly.Internal.Data.SVar (adaptState)
-import Control.Monad (when)
 import Streamly.Internal.Data.Strict (Tuple'(..))
 
-
 import qualified Streamly.Internal.Data.Stream.StreamD.Type as D
 
+import GHC.Exts
+import Prelude hiding (length, unlines)
+
 -------------------------------------------------------------------------------
 -- Array Data Type
 -------------------------------------------------------------------------------
 
+-- XXX why is this not supposed to be used?
```

Comment:
I've written this at the beginning with something in mind about a clean data type.